enhancements/kata-containers/kata-containers-dev-preview.md

```diff
@@ -0,0 +1,260 @@
+
+See additional details in: https://github.com/openshift/enhancements/pull/317
+
+**So What do we want to do with this?**
+
+We start by creating a container image with only the kata upstream components as static binaries (no RHEL-AV components). The kata operator will create a container using this image. By doing this the kata binaries can be accessed by the host without copying files to the host filesystem. It would be just another container running with no pollution of the RHCOS installation. 
+We then use the RHCOS extensions for deploying the RHEL-AV RPMs. 
+
+**Pros:**
+1. RHEL-AV depends on libraries and other tools which are all available as RPMs. Turning this into a static image instead of RPMs creates a huge single binary which we avoid with extensions
+2. RHEL-AV has been tested / validated as RPMs and not a single binary which means we saves the extra overhead 
+3. RHEL-AV RPMs can be installed only by the customers who would be interested in using the kata runtime. QEMU (and its dependencies) get updated as part of the machine-os-content updates
+4. There's no extra work related to properly loading qemu-kvm-core dependencies, as all the packages would be installed in the same PATH its build in
+5. Better control of the versions that are installed both for qemu (specific version for a given host) and for the dependencies (tracked by rpm)
+
+**Cons:**
+1. ~75 mb extra, installed in the host, for those who'd be using kata runtime
+2. machine-os-content would still carry this until there is a separate machine-os-content- extensions container
+
+This approach has a dependency on the extensions framework being delivered into Openshift.
+
+
+### KataContainer Operator development 
+The goal is to develop a kata operator that can be used to manage the entire lifecycle of kata components in an OpenShift cluster. We will have a controller that watches for a kata custom resource and a daemon-set that acts upon changes to the CR and do the required work on the worker nodes (install, uninstall update,...). 
+
+Via the CR it will be possible to select a subset of worker nodes, an image that contains all the binaries and optionally the name of the runtime class (If the runtime class is not defined, the name of the CR is used to create a runtime class configuration in crio). For deploying binaries on the host the current idea is to use a container image that can be mounted on the host level.  The goal is to be able to install the operator via OperatorHub. The operator will also create a crio drop-in config file via machine config objects. 
```

Comment:
```suggestion
Via the CR it will be possible to select a subset of worker nodes. For deploying binaries on the host the current idea is to use a container image that can be mounted on the host level.  The goal is to be able to install the operator via OperatorHub. The operator will also create a crio drop-in config file via machine config objects. The Operator will automatically select the payload image that contains correct version of the kata binaries for the given version of the OpenShift. The Operator will also configure a `RuntimeClass` called `kata-oc` which can be used to deploy workload that uses kata runtime. 

The `RuntimeClass` and payload image names will be visible in the CR status. 
```

Reply:
@mrunalp @ariel-adam please review.

Reply:
Fixed.